src/engine/EventManager.cxx

```diff
@@ -562,14 +556,20 @@ void EventManager::checkEvents(SDL_Event &event, Engine &engine)
         else if (!tileToPlace.empty() && m_placementAllowed)
         {
           // if targetObject.size > 1 it is a tile bigger than 1x1
+            // TODO: Remove the check here, there is no need for handling this differently
           if (targetObjectNodes.size() > 1 && isPointWithinMapBoundaries(targetObjectNodes))
           {
             // instead of using "nodesToPlace" which would be the origin-corner coordinate, we need to pass ALL occupied nodes for now.
-            engine.setTileIDOfNode(targetObjectNodes.begin(), targetObjectNodes.end(), tileToPlace, false);
+            //engine.setTileIDOfNode(targetObjectNodes.begin(), targetObjectNodes.end(), tileToPlace, false);
+            // TODO: If a building is selected, it still allows for line seelction.  m_nodesToPlace would contain more buildings figure out what this is.
+            //engine.map->setTileID(tileToPlace, mouseIsoCoords); // use this for now to workaround wrongly allowed line selection
+            engine.map->setTileID(tileToPlace, m_nodesToPlace);
+
           }
           else
           {
-            engine.setTileIDOfNode(m_nodesToPlace.begin(), m_nodesToPlace.end(), tileToPlace, true);
+            engine.map->setTileID(tileToPlace, m_nodesToPlace);
+            //engine.setTileIDOfNode(m_nodesToPlace.begin(), m_nodesToPlace.end(), tileToPlace, true);
```

Comment:
I think we can remove this line

Reply:
the entire function will be removed once the other PRs are merge. setTileID is used in ZoneManager and this has changed so much in the other PR that it would be merge hell if I change it before